python/pylibcudf/pylibcudf/io/parquet.pyx

```diff
@@ -221,3 +244,332 @@ cpdef read_parquet(
+        enabled : bool
+            Boolean value to enable/disable int96 timestamps
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.int96_timestamps(enabled)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder write_v2_headers(self, bool enabled):
+        """
+        Set to true if V2 page headers are to be written.
+
+        Parameters
+        ----------
+        enabled : bool
+            Boolean value to enable/disable writing of V2 page headers.
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.write_v2_headers(enabled)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder dictionary_policy(self, dictionary_policy_t val):
+        """
+        Sets the policy for dictionary use.
+
+        Parameters
+        ----------
+        val : DictionaryPolicy
+            Policy for dictionary use.
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.dictionary_policy(val)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder utc_timestamps(self, bool enabled):
+        """
+        Set to true if timestamps are to be written as UTC.
+
+        Parameters
+        ----------
+        enabled : bool
+            Boolean value to enable/disable writing of timestamps as UTC.
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.utc_timestamps(enabled)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder write_arrow_schema(self, bool enabled):
+        """
+        Set to true if arrow schema is to be written.
+
+        Parameters
+        ----------
+        enabled : bool
+            Boolean value to enable/disable writing of arrow schema.
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.write_arrow_schema(enabled)
+        return self
+
+    cpdef ParquetWriterOptions build(self):
+        """
+        Options member once it's built
+
+        Returns
+        -------
+        ParquetWriterOptions
+        """
+        cdef ParquetWriterOptions parquet_options = ParquetWriterOptions.__new__(
+            ParquetWriterOptions
+        )
+        parquet_options.c_obj = move(self.c_obj.build())
+        return parquet_options
+
+
+cpdef HostBuffer write_parquet(ParquetWriterOptions options):
+    """
+    Writes a set of columns to parquet format.
+
+    Parameters
+    ----------
+    options : ParquetWriterOptions
+        Settings for controlling writing behavior
+
+    Returns
+    -------
+    pylibcudf.contiguous_split.HostBuffer
+        A blob that contains the file metadata
+        (parquet FileMetadata thrift message) if requested in
+        parquet_writer_options (empty blob otherwise).
+    """
+    cdef parquet_writer_options c_options = options.c_obj
+    cdef unique_ptr[vector[uint8_t]] c_result
+
+    with nogil:
+        c_result = cpp_write_parquet(c_options)
+
+    return HostBuffer.from_unique_ptr(move(c_result))
```

Comment:
*suggestion*: Return a `memoryview` rather than a `HostBuffer`? I guess it doesn't really matter too much.